### PR TITLE
E2E test case adding a new system user

### PR DESCRIPTION
**Pull request by spyrkob**
Works towards fixing #4023

#### Short description of what this resolves:
Adding new e2e test

#### Changes proposed in this pull request:
Test case covering adding new system user. Adding initial [Page Objects]( https://www.automatetheplanet.com/page-object-pattern/)to help with writing more e2e tests.

### CODE REVIEW

**Comment by bradymiller** [comment]
hi @spyrkob , Looks very promising! Just let us know when you want a code review. thanks! -brady

**Comment by spyrkob** [comment]
Hi @bradymiller I think it should be ready for review. I'll open new PRs for further tests, so would be good to know if there's any problems with this one :)

File: tests/Tests/E2e/CreateStaffTest.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+use Symfony\Component\Panther\Client;
+use OpenEMR\Tests\E2e\Pages\{LoginPage, MainPage};
+
+class CreateStaffTest extends PantherTestCase
+{
+    private $e2eBaseUrl;
+
+    protected function setUp(): void
+    {
+        $this->e2eBaseUrl = getenv("OPENEMR_BASE_URL", true) ?: "http://localhost";
```
Comment (bradymiller) [inline tests/Tests/E2e/CreateStaffTest.php:17]: note that when you rebase/merge in current codebase, this variable is now (as of last night :) ): 
OPENEMR_BASE_URL_E2E

File: tests/Tests/E2e/ExampleE2eTest.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+class ExampleE2eTest extends PantherTestCase
+{
+    /**
+     * The base url used for e2e (end to end) browser testing.
+     */
+    private $e2eBaseUrl;
+
+    protected function setUp(): void
+    {
+        $this->e2eBaseUrl = getenv("OPENEMR_BASE_URL", true) ?: "http://localhost";
```
Comment (bradymiller) [inline tests/Tests/E2e/ExampleE2eTest.php:19]: note that when you rebase/merge in current codebase, this variable is now (as of last night :) ): 
OPENEMR_BASE_URL_E2E

File: tests/Tests/E2e/ExampleE2eTest.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    public function visitor_with_valid_credential_can_be_authenticated(): void
+    {
+        $openEmrPage = $this->e2eBaseUrl;
+        // ok - PantherClient
+        $client = static::createPantherClient(['external_base_uri' => $openEmrPage]);
+        $crawler = $client->request('GET', '/interface/login/login.php?site=default');
+
+        $v = $client->getCurrentURL();
+
+        echo $v;
```
Comment (bradymiller) [inline tests/Tests/E2e/ExampleE2eTest.php:32]: is this echo part of the test?

File: tests/Tests/E2e/ExampleE2eTest.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?php
```
Comment (bradymiller) [inline tests/Tests/E2e/ExampleE2eTest.php:1]: this script looks like a partial duplicate of tests/Tests/E2e/TestOpenemrPage/ExampleE2eTest.php . Is there a reason for this (ie. migrating test location maybe?)?

**Comment by bradymiller** [comment]
hi @spyrkob , Really nice stuff; the Page Objects are especially nice and see huge potential for these as develop more tests. Only noted several minor issues/questions which I posted above. thanks! -brady

File: tests/Tests/E2e/ExampleE2eTest.php (hunk 4 above)
Comment (spyrkob) [reply tests/Tests/E2e/ExampleE2eTest.php:1]: Yes that script should not be there. I had some issues running e2e issues and was messing with files, this must have survived the cleanup... I'll remove this file.

**Comment by spyrkob** [comment]
Thanks @bradymiller :) The comments should be addressed now - I removed the ExampleE2eTest.php and changed the property name

**Comment by bradymiller** [comment]
hi @spyrkob , Code looks good and bringing in. Thanks for the improvements! -brady

**Pull request merged**